git-issue.sh

```diff
@@ -855,26 +855,97 @@ usage_list()
 {
   cat <<\USAGE_list_EOF
 gi new usage: git issue list [-a] [tag|milestone]
+              git issue list [-a] -l formatstring [-o sort_field] [-r] [tag|milestone]
 USAGE_list_EOF
   exit 2
 }
 
+# helper function for long listing format, each call proccesses a single issue
+shortshow()
+{
+
+  local Aid Adate Amilestone Aassignee Atags Adescription
+  #SHA id
+  Aid=$id
+  # Date
+  Adate=`git show --no-patch --format='%ai' $id`
+
+  # Milestone
+  if [ -s $path/milestone ] ; then
+    Amilestone=`fmt $path/milestone`
+  fi
+
+  # Assignee
+  if [ -r $path/assignee ] ; then
+    Aassignee=`fmt $path/assignee`
+  fi
+
+  # Tags
+  if [ -s $path/tags ] ; then
+    Atags=`fmt $path/tags`
+  fi
+
+  # Description
+  Adescription=`head -n 1 $path/description` 
+
+
+  (echo -nE "$sortfield";printf '\002';echo -E "$formatstring") | #we print the field to sort by first, and remove it after sorting 
+  sed -e s$'\003'%n$'\003'"\n"$'\003'g \
+  -e s$'\003'%i$'\003'"$Aid"$'\003'g \
+  -e s$'\003'%c$'\003'"$Adate"$'\003'g \
+  -e s$'\003'%M$'\003'"$Amilestone"$'\003'g \
+  -e s$'\003'%A$'\003'"$Aassignee"$'\003'g \
+  -e s$'\003'%T$'\003'"$Atags"$'\003'g \
+  -e s$'\003'%D$'\003'"$Adescription"$'\003'g | #an nonprintable char is used as delimiter since the fields can contain any char
+  tr "\n" '\001' >> results
```

Comment:
Don't append do the file. Instead write to stdout, and have a single tee capture the output. (See below.)